wasm/module.go

```diff
@@ -78,3 +80,61 @@ func DecodeModule(binary []byte) (*Module, error) {
 	}
 	return ret, nil
 }
+
+func (m *Module) GetFunctionNames() (map[uint32]string, error) {
+	namesec, ok := m.CustomSections["name"]
+	if !ok {
+		return nil, fmt.Errorf("'name' %w", ErrCustomSectionNotFound)
+	}
+
+	r := bytes.NewReader(namesec)
+	for {
+		id, err := r.ReadByte()
+		if err != nil {
+			return nil, fmt.Errorf("failed to read subsection ID: %w", err)
+		}
+
+		size, _, err := leb128.DecodeUint32(r)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read the size of subsection: %w", err)
+		}
+
+		if id == 1 {
+			// ID = 1 is the function name subsection.
+			break
+		} else {
+			// Skip other subsections.
+			_, err := r.Seek(int64(size), io.SeekCurrent)
+			if err != nil {
+				return nil, fmt.Errorf("failed to seek: %w", err)
```

Comment:
maybe error should be like the other.. `"failed to skip subsection %s: %w", id, err`